python/ray/data/dataset.py

```diff
@@ -282,6 +282,8 @@ def map(
         fn: UserDefinedFunction[Dict[str, Any], Dict[str, Any]],
         *,
         compute: Optional[ComputeStrategy] = None,
+        num_cpus: Optional[Union[int, float]] = None,
+        num_gpus: Optional[Union[int, float]] = None,
```

Comment:
```suggestion
        num_cpus: Optional[float] = None,
        num_gpus: Optional[float] = None,
```

> Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: **when an argument is annotated as having type float, an argument of type int is acceptable**; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable.
- [PEP 484 Type Hints](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower)

Reply:
Ok sounds good...I copied this from the Ray Core type hints

Reply:
updated